File: swift/Sources/Schemas/CreateManagedAgentSourceRequestSourceKind.swift
import Foundation

public enum CreateManagedAgentSourceRequestSourceKind: String, Codable, Hashable, CaseIterable, Sendable {
    case workflowJson = "workflow_json"
    case langgraph
    case n8NJson = "n8n_json"
    case yamlDag = "yaml_dag"
    case sourceBundle = "source_bundle"
    case repoRef = "repo_ref"
    case containerRef = "container_ref"
    case adapterEndpoint = "adapter_endpoint"
    case promptBundle = "prompt_bundle"
    case skillTree = "skill_tree"
    case memoryProfile = "memory_profile"
    case evalSuite = "eval_suite"
    case traceBundle = "trace_bundle"
    case naturalLanguage = "natural_language"
}

File: ruby/lib/Sikaru/types/create_managed_agent_source_request_source_kind.rb
# frozen_string_literal: true

module Sikaru
  module Types
    module CreateManagedAgentSourceRequestSourceKind
      extend Sikaru::Internal::Types::Enum

      WORKFLOW_JSON = "workflow_json"
      LANGGRAPH = "langgraph"
      N8N_JSON = "n8n_json"
      YAML_DAG = "yaml_dag"
      SOURCE_BUNDLE = "source_bundle"
      REPO_REF = "repo_ref"
      CONTAINER_REF = "container_ref"
      ADAPTER_ENDPOINT = "adapter_endpoint"
      PROMPT_BUNDLE = "prompt_bundle"
      SKILL_TREE = "skill_tree"
      MEMORY_PROFILE = "memory_profile"
      EVAL_SUITE = "eval_suite"
      TRACE_BUNDLE = "trace_bundle"
      NATURAL_LANGUAGE = "natural_language"
    end
  end
end
